src/Emogrifier/CssInliner.php

```diff
@@ -662,14 +663,46 @@ function (array $cssRule) {
      */
     private function removeUnmatchablePseudoComponents($selector)
     {
+        // The regex allows nested brackets via `(?2)`.
+        // A space is temporarily prepended because the callback can't determine if the match was at the very start.
+        $selectorWithoutNots = \ltrim(\preg_replace_callback(
+            '/(\\s?+):not(\\([^\\(\\)]*+(?:(?2)[^\\(\\)]*+)*+\\))/i',
+            [$this, 'replaceUnmatchableNotComponent'],
+            ' ' . $selector
+        ));
+
         $pseudoComponentMatcher = ':(?!' . static::PSEUDO_CLASS_MATCHER . '):?+[\\w\\-]++(?:\\([^\\)]*+\\))?+';
         return \preg_replace(
             ['/(\\s|^)' . $pseudoComponentMatcher . '/i', '/' . $pseudoComponentMatcher . '/i'],
             ['$1*', ''],
-            $selector
+            $selectorWithoutNots
         );
     }
 
+    /**
+     * Helper for `removeUnmatchablePseudoComponents()` to replace/remove a selector `:not` component if its argument
+     * contains pseudo-elements or dynamic pseudo-classes.
+     *
+     * @param string[] $matches Array of capturing groups matched by the regular expression.
+     *        Index 0 contains the full match ":not(...)", preceded by a whitespace character if present.
+     *        Index 1 contains any preceding whitespace character or an empty string if none present.
+     *        Index 2 contains the `:not` argument with the surrounding brackets (i.e. "(...)").
```

Comment:
Instead of explaining the indices in the comment, I propose we use local variables in the methos.

Reply:
Done.